dll/win32/shlwapi/ordinal.c

```diff
@@ -5322,42 +5322,154 @@ HRESULT WINAPI SHPropertyBag_ReadLONG(IPropertyBag *ppb, LPCWSTR pszPropName, LP
+        return E_INVALIDARG;
+    }
+
+    V_VT(&vari) = VT_BOOL;
+    V_BOOL(&vari) = (bValue ? VARIANT_TRUE : VARIANT_FALSE); /* NOTE: VARIANT_TRUE is (SHORT)-1 */
+    return IPropertyBag_Write(ppb, pszPropName, &vari);
+}
+
+/**************************************************************************
+ *  SHPropertyBag_WriteSHORT (SHLWAPI.528)
+ */
+HRESULT WINAPI SHPropertyBag_WriteSHORT(IPropertyBag *ppb, LPCOLESTR pszPropName, SHORT sValue)
+{
+    VARIANT vari;
+
+    TRACE("%p %s %d\n", ppb, debugstr_w(pszPropName), sValue);
+
+    if (!ppb || !pszPropName)
+    {
+        ERR("%p %s\n", ppb, debugstr_w(pszPropName));
+        return E_INVALIDARG;
+    }
+
+    V_VT(&vari) = VT_UI2;
+    V_UI2(&vari) = sValue;
+    return IPropertyBag_Write(ppb, pszPropName, &vari);
+}
+
 /**************************************************************************
  *  SHPropertyBag_WriteLONG (SHLWAPI.497)
  *
  * This function asks a property bag to write a named property as a LONG.
- *
- * PARAMS
- *  ppb: a IPropertyBag interface
- *  pszPropName:  Unicode string that names the property
- *  lValue: address to receive the property value as a 32-bit signed integer
- *
- * RETURNS
- *  HRESULT codes
  */
-HRESULT WINAPI SHPropertyBag_WriteLONG(IPropertyBag *ppb, LPCWSTR pszPropName, LONG lValue)
+HRESULT WINAPI SHPropertyBag_WriteLONG(IPropertyBag *ppb, LPCOLESTR pszPropName, LONG lValue)
 {
-	UNIMPLEMENTED;
-	return E_NOTIMPL;
+    VARIANT vari;
+
+    TRACE("%p %s %ld\n", ppb, debugstr_w(pszPropName), lValue);
+
+    if (!ppb || !pszPropName)
+    {
+        ERR("%p %s\n", ppb, debugstr_w(pszPropName));
+        return E_INVALIDARG;
+    }
+
+    V_VT(&vari) = VT_I4;
+    V_I4(&vari) = lValue;
+    return IPropertyBag_Write(ppb, pszPropName, &vari);
+}
+
+/**************************************************************************
+ *  SHPropertyBag_WriteDWORD (SHLWAPI.508)
+ */
+HRESULT WINAPI SHPropertyBag_WriteDWORD(IPropertyBag *ppb, LPCOLESTR pszPropName, DWORD dwValue)
+{
+    VARIANT vari;
+
+    TRACE("%p %s %lu\n", ppb, debugstr_w(pszPropName), dwValue);
+
+    if (!ppb || !pszPropName)
+    {
+        ERR("%p %s\n", ppb, debugstr_w(pszPropName));
+        return E_INVALIDARG;
+    }
+
+    V_VT(&vari) = VT_UI4;
+    V_UI4(&vari) = dwValue;
+    return IPropertyBag_Write(ppb, pszPropName, &vari);
 }
 
 /**************************************************************************
  *  SHPropertyBag_WriteStr (SHLWAPI.495)
  *
  * This function asks a property bag to write a string as the value of a named property.
- *
- * PARAMS
- *  ppb: a IPropertyBag interface
- *  pszPropName:  Unicode string that names the property
- *  pValue: address to write the property value
- *
- * RETURNS
- *  HRESULT codes
  */
-HRESULT WINAPI SHPropertyBag_WriteStr(IPropertyBag *ppb, LPCWSTR pszPropName, LPCWSTR pszValue)
+HRESULT WINAPI SHPropertyBag_WriteStr(IPropertyBag *ppb, LPCOLESTR pszPropName, LPCWSTR pszValue)
```

Comment:
Huh.... why `LPCOLESTR` instead of `LPCWSTR` ?

Reply:
In some old systems, `OLECHAR` type is `CHAR`.

Reply:
yes, but the same as TCHAR...

Those APIs use unicode strings, so they should keep the LPCWSTR instead of changing the parameter.

Reply:
Fixed.